src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineInstance.java

```diff
@@ -84,21 +90,65 @@ protected void _terminate(TaskListener listener) throws IOException, Interrupted
         try {
             ComputeEngineCloud cloud = getCloud();
 
-            Computer computer = this.toComputer();
-            if (this.oneShot && this.createSnapshot && computer != null && !computer.getBuilds().failureOnly().isEmpty()) {
-                LOGGER.log(Level.INFO, "Creating snapshot for node ... " + this.getNodeName());
-                cloud.getClient().createSnapshot(cloud.projectId, this.zone, this.getNodeName());
-            }
+            ExecutorService executor = Executors.newFixedThreadPool(1);
 
-            // If the instance is running, attempt to terminate it. This is an asynch call and we
-            // return immediately, hoping for the best.
-            cloud.client.terminateInstanceWithStatus(cloud.projectId, zone, name, "RUNNING");
+            executor.submit(new TerminateRunnable(this, cloud));
         } catch (CloudNotFoundException cnfe) {
             listener.error(cnfe.getMessage());
             return;
         }
 
+    }
+
+    /**
+     * Runnable to parallelize the termination and snapshot creation process.
+     */
+    private class TerminateRunnable implements Runnable {
+        private ComputeEngineCloud cloud;
+        private Computer computer;
+        private ComputeEngineInstance computeEngineInstance;
+
+        /**
+         * Constructor.
+         *
+         * @param computeEngineInstance
+         * @param cloud
+         */
+        TerminateRunnable(ComputeEngineInstance computeEngineInstance, ComputeEngineCloud cloud) {
+            this.cloud = cloud;
+            this.computer = computeEngineInstance.toComputer();
+            this.computeEngineInstance = computeEngineInstance;
+        }
 
+        /**
+         * Checks to see if snapshot creation at deletion time requirements are fulfilled and creates if true.
+         * Terminates the instance afterwards.
+         */
+        @Override
+        public void run() {
+            String projectId = cloud.getProjectId();
+            if (computeEngineInstance.oneShot && computeEngineInstance.createSnapshot && computer != null &&
+                    !computer.getBuilds().failureOnly().isEmpty()) {
+                LOGGER.log(Level.INFO, "Creating snapshot for node " + name);
+                try {
+                    cloud.getClient().createSnapshot(projectId, zone, name);
+                } catch (IOException | InterruptedException e) {
+                    // exceptions thrown by snapshot creation are InterruptedException and IOException
+                    LOGGER.log(Level.WARNING, "Error creating snapshot for instance " + name + ". " + e);
+                }
+            }
+
+            // If the instance is running, attempt to terminate it. This is an asynch call and we
```

Comment:
Remove/change not correct now comment

Reply:
The logic below is now synchronous, this comment block should be updated to reflect.